plasmapy/classes/sources/tests/test_openpmd_hdf5.py

```diff
@@ -0,0 +1,76 @@
+from plasmapy.classes.sources import openpmd_hdf5
+from plasmapy.utils import OpenPMDError
+from plasmapy.data.test import rootdir
+
+from astropy import units as u
+from typing import Union, Tuple, List
+import os
+import pytest
+
+
+class TestOpenPMD2D:
+    """Test 2D HDF5 dataset based on OpenPMD."""
+    # Downloaded from
+    # https://github.com/openPMD/openPMD-example-datasets/blob/draft/example-2d.tar.gz
+    # per the Creative Commons Zero v1.0 Universal license
+    h5 = openpmd_hdf5.HDF5Reader(hdf5=os.path.join(rootdir, "data00000255.h5"))
+
+    def test_has_electric_field_with_units(self):
+        assert self.h5.electric_field.to(u.V / u.m)
+
+    def test_correct_shape_electric_field(self):
+        assert self.h5.electric_field.shape == (3, 51, 201)
+
+    def test_has_charge_density_with_units(self):
+        assert self.h5.charge_density.to(u.C / u.m**3)
+
+    def test_correct_shape_charge_density(self):
+        assert self.h5.charge_density.shape == (51, 201)
+
+
+class TestOpenPMD3D:
+    """Test 3D HDF5 dataset based on OpenPMD."""
+    # Downloaded from
+    # https://github.com/openPMD/openPMD-example-datasets/blob/draft/example-3d.tar.gz
+    # per the Creative Commons Zero v1.0 Universal license
+    h5 = openpmd_hdf5.HDF5Reader(hdf5=os.path.join(rootdir, "data00000100.h5"))
+
+    def test_has_electric_field_with_units(self):
+        assert self.h5.electric_field.to(u.V / u.m)
+
+    def test_correct_shape_electric_field(self):
+        assert self.h5.electric_field.shape == (3, 26, 26, 201)
```

Comment:
I might mess around with this one so that the interface handles better in a soon-to-be-made PR.